Source/WebCore/platform/graphics/cocoa/VideoMediaSampleRenderer.mm

```diff
@@ -1112,6 +1112,24 @@ static bool isRendererThreadSafe(WebSampleBufferVideoRendering *renderering)
     m_resourceOwner = resourceOwner;
 }
 
+void VideoMediaSampleRenderer::invalidateDecompressionSession()
+{
+    dispatcher()->dispatch([weakThis = ThreadSafeWeakPtr { *this }] {
+        RefPtr protectedThis = weakThis.get();
+        if (!protectedThis)
+            return;
+
+        RefPtr decompressionSession = [&] {
+            Locker lock { protectedThis->m_lock };
+            return std::exchange(protectedThis->m_decompressionSession, nullptr);
+        }();
+
+        if (decompressionSession)
+            decompressionSession->invalidate();
+        protectedThis->notifyVideoRendererRequiresFlushToResumeDecoding();
```

Comment:
Shouldn't we hop to main thread here?
Also maybe we could have a routine that does the exchange/invalidation and notification, like done in case of `kVTInvalidSessionErr`.